src/qengine/opencl.cpp

```diff
@@ -979,12 +981,12 @@ void QEngineOCL::DecomposeDispose(bitLenInt start, bitLenInt length, QEngineOCLP
     size_t ngs = FixGroupSize(ngc, nrmGroupSize);
 
     // The "remainder" bits will always be maintained.
-    real1* remainderStateProb = new real1[remainderPower]();
-    real1* remainderStateAngle = new real1[remainderPower]();
+    std::shared_ptr<real1> remainderStateProb(new real1[remainderPower](), [](real1* p) { delete[] p; });
```

Comment:
Too bad this isn't C++17 compat, which doesn't require these workarounds.

Reply:
Awesome that we can keep C++11 compatibility, with only this one workaround. :-)